bundle/terranova/apply.go

```diff
@@ -65,24 +66,34 @@ func (d *DeploymentUnit) Deploy(ctx context.Context, db *tnstate.TerranovaState,
 }
 
 func (d *DeploymentUnit) Create(ctx context.Context, db *tnstate.TerranovaState, config any) error {
-	newID, err := d.Adapter.DoCreate(ctx, config)
+	newID, remoteState, err := d.Adapter.DoCreate(ctx, config)
 	if err != nil {
 		// No need to prefix error, there is no ambiguity (only one operation - DoCreate) and no additional context (like id)
 		return err
 	}
 
 	log.Infof(ctx, "Created %s.%s id=%#v", d.Group, d.Key, newID)
 
+	err = d.SetRemoteState(remoteState)
+	if err != nil {
+		return err
+	}
+
 	err = db.SaveState(d.Group, d.Key, newID, config)
 	if err != nil {
 		return fmt.Errorf("saving state after creating id=%s: %w", newID, err)
 	}
 
-	err = d.Adapter.WaitAfterCreate(ctx, config)
+	waitRemoteState, err := d.Adapter.WaitAfterCreate(ctx, config)
 	if err != nil {
 		return fmt.Errorf("waiting after creating id=%s: %w", newID, err)
 	}
 
+	err = d.SetRemoteState(waitRemoteState)
+	if err != nil {
+		return err
+	}
+
```

Comment:
`DoCreate` may return a `nil` remote state or not. It is not clear when it is guaranteed to be populated.